modules/openapi-generator/src/main/java/org/openapitools/codegen/InlineModelResolver.java

```diff
@@ -104,95 +105,220 @@ private void flattenPaths(OpenAPI openAPI) {
+            // Check properties and recurse, each property could be its own inline model
+            Map<String, Schema> props = schema.getProperties();
+            if (props != null) {
+                for (String propName : props.keySet()) {
+                    Schema prop = props.get(propName);
+                    // Recurse to create $refs for inner models
+                    gatherInlineModels(prop, modelPrefix + StringUtils.camelize(propName));
+                    if (isModelNeeded(prop)) {
+                        // If this schema should be split into its own model, do so
+                        Schema refSchema = this.makeSchemaResolve(modelPrefix, StringUtils.camelize(propName), prop);
+                        props.put(propName, refSchema);
                     }
-
-                    for (String consume : consumes) {
-                        content.addMediaType(consume, mt);
+                }
+            }
+            // Check additionalProperties for inline models
+            if (schema.getAdditionalProperties() != null) {
+                if (schema.getAdditionalProperties() instanceof Schema) {
+                    Schema inner = (Schema) schema.getAdditionalProperties();
+                    // Recurse to create $refs for inner models
+                    gatherInlineModels(inner, modelPrefix + "AddlProps");
+                    if (isModelNeeded(inner)) {
+                        // If this schema should be split into its own model, do so
+                        Schema refSchema = this.makeSchemaResolve(modelPrefix, "AddlProps", inner);
+                        schema.setAdditionalProperties(refSchema);
                     }
-
-                    rb.setContent(content);
-
-                    // add to openapi "components"
-                    if (openAPI.getComponents().getRequestBodies() == null) {
-                        Map<String, RequestBody> requestBodies = new HashMap<String, RequestBody>();
-                        requestBodies.put(modelName, rb);
-                        openAPI.getComponents().setRequestBodies(requestBodies);
+                }
+            }
+        } else if (schema.getProperties() != null) {
+            // If non-object type is specified but also properties
+            LOGGER.error("Illegal schema found with non-object type combined with properties," +
+                    " no properties should be defined:\n " + schema.toString());
+            return;
+        } else if (schema.getAdditionalProperties() != null) {
+            // If non-object type is specified but also additionalProperties
+            LOGGER.error("Illegal schema found with non-object type combined with" +
+                    " additionalProperties, no additionalProperties should be defined:\n " +
+                    schema.toString());
+            return;
+        }
+        // Check array items
+        if (schema instanceof ArraySchema) {
```

Comment:
Arraymodels are generated in some generators. Are we covering that use case?

Reply:
The main affect of this PR for Array models would be to promote consistency such that the same models get generated regardless of whether they appear via inline or as a $ref
e.g. making sure this inline spec
```yaml
        kittens:
          type: array
          items: # inline array where the items are enums
            type: string
            enum:
              - Kitten1
              - Kitten2
              - Kitten3
```
generates the same as
```yaml
        kittens:
          type: array
          items:
            $ref: '#/components/schemas/Cat'
```

In general the ArrayModels should be handled the same after this PR as before it unless there was particular logic to handle complex inline array models such as the above (I have seen languages like Typescript handle inline enum generation using complex template logic for example that should no longer be needed).

If you have specific languages in mind that you're concerned about for backwards compatibility I'm happy to take a look. The most likely impact would be a change to the name of the inline model to be more specific e.g. `OASInlineObject1` -> `OASUploadFileBody` or completely ungenerated models now being generated.

Reply:
Thank you for explaining this